src/DurableTask.Core/TaskEntityDispatcher.cs

```diff
@@ -801,8 +801,8 @@ void ProcessSendStartMessage(WorkItemEffects effects, OrchestrationRuntimeState
             var executionStartedEvent = new ExecutionStartedEvent(-1, action.Input)
             {
                 Tags = OrchestrationTags.MergeTags(
-                    runtimeState.Tags,
-                    new Dictionary<string, string>() { { OrchestrationTags.FireAndForget, "" } }),
+                    new Dictionary<string, string>() { { OrchestrationTags.FireAndForget, "" } },
+                    runtimeState.Tags),
```

Comment:
To prevent future error, let's add explicit the parameter names here, so that we know which tags are supposed to be the new ones, and which are the old ones:

```suggestion
                    newTags: new Dictionary<string, string>() { { OrchestrationTags.FireAndForget, "" } },
                    existingTags: runtimeState.Tags),
```